library/objective-c/EnvoyStream.h

```diff
@@ -0,0 +1,56 @@
+#import <Foundation/Foundation.h>
+
+#import "library/objective-c/EnvoyHeaders.h"
+#import "library/objective-c/EnvoyObserver.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface EnvoyStream : NSObject
+
+/**
+ Open an underlying HTTP stream.
+
+ @param observer the observer that will run the stream callbacks.
+ */
+- (instancetype)initWithObserver:(EnvoyObserver *)observer;
+
+/**
+ Send headers over the provided stream.
+
+ @param headers Headers to send over the stream.
+ @param close True if the stream should be closed after sending.
+ */
+- (void)sendHeaders:(EnvoyHeaders *)headers close:(BOOL)close;
+
+/**
+ Send data over the provided stream.
+
+ @param data Data to send over the stream.
+ @param close True if the stream should be closed after sending.
+ */
+- (void)sendData:(NSData *)data close:(BOOL)close;
+
+/**
+ Send metadata over the provided stream.
+
+ @param metadata Metadata to send over the stream.
+ */
+- (void)sendMetadata:(EnvoyHeaders *)metadata;
```

Comment:
We should probably fix these inconsistencies between the different methods in the `main_interface.h` because not all of the send methods have the option of closing/ending the streaming

Reply:
Metadata can't close the stream; it's unsupported by the protocol. For now we'll be closing all streams by sending an empty data frame.